File: Examples/WordFrequency/WordFrequency.cs
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.0

using System;
using System.Collections.Generic;
using System.Linq;

namespace Examples.WordFrequency;

public static class WordFrequency
{
    public static void Main()
    {
        string[] words = ["visual", "xsharp", "visual", "compiler", "xsharp", "visual"];
        Dictionary<string, int> counts = [];

        foreach (var word in words)
        {
            counts[word] = counts.GetValueOrDefault(word) + 1;
        }

        foreach (var (word, count) in counts.OrderBy(pair => pair.Key))
        {
            Console.WriteLine($"{word}: {count}");
        }
    }
}

File: Examples/WordFrequency/WordFrequency.cpp
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.0

#include <array>
#include <iostream>
#include <map>
#include <string>

int
main()
{
    constexpr std::array words = { "visual", "xsharp", "visual", "compiler", "xsharp", "visual" };
    std::map<std::string, int> counts;
    for (const auto *word : words)
    {
        ++counts[word];
    }
    for (const auto &[word, count] : counts)
    {
        std::cout << word << ": " << count << '\n';
    }
}
